### PR TITLE
Kconfig: Expose gnrc/ipv6/whitelist configurations

### PR DESCRIPTION
**Pull request by leandrolanzieri**
### Contribution description
This moves the `gnrc_ipv6_whitelist` module configuration macros to the `CONFIG_` namespace and exposes them to Kconfig.

### Testing procedure
- Compile `gnrc_networking` example application using this module (`USEMODULE=gnrc_ipv6_whitelist make all term`), the default configuration should still be the same.

- Call menuconfig using this module (`USEMODULE=gnrc_ipv6_whitelist make menuconfig`), you should the list size as an option. The default list size should still be the same. If you change the value it should change in the app.

### Issues/PRs references
Part of #12888

### CODE REVIEW

File: sys/include/net/gnrc/ipv6/whitelist.h [hunk 1]
```diff
@@ -37,8 +37,8 @@ extern "C" {
 /**
  * Maximum size of the whitelist.
  */
-#ifndef GNRC_IPV6_WHITELIST_SIZE
-#define GNRC_IPV6_WHITELIST_SIZE    (8)
+#ifndef CONFIG_GNRC_IPV6_WHITELIST_SIZE
+#define CONFIG_GNRC_IPV6_WHITELIST_SIZE    (8)
```
Comment (miri64) [inline sys/include/net/gnrc/ipv6/whitelist.h]: Do we *have* to rename all config macros to be usable with kconfig?

Comment (leandrolanzieri) [reply sys/include/net/gnrc/ipv6/whitelist.h]: Yes, Kconfig generates the .config and .h files appending the `CONFIG_` symbol. Regardless of that, I think it's good to have configuration macros under a single namespace.

Comment (miri64) [reply sys/include/net/gnrc/ipv6/whitelist.h]: Ok, were just wondering for future integrations.

**Comment by jia200x** [comment]
I set the whitelist size to 100 via menuconfig and applied the following patch:
```
diff --git a/examples/gnrc_networking/main.c b/examples/gnrc_networking/main.c
index 6301f4291..c9b4b9efe 100644
--- a/examples/gnrc_networking/main.c
+++ b/examples/gnrc_networking/main.c
@@ -39,6 +39,7 @@ int main(void)
      * receive potentially fast incoming networking packets */
     msg_init_queue(_main_msg_queue, MAIN_QUEUE_SIZE);
     puts("RIOT network stack example application");
+    printf("%i\n", CONFIG_GNRC_IPV6_WHITELIST_SIZE);
 
     /* start shell */
     puts("All up, running the shell now");
```

The output is
```
main(): This is RIOT! (Version: 2020.01-devel-1311-g9a0a51-copr/12893)
RIOT network stack example application
100
All up, running the shell now
```

So, it works as expected.

**Comment by jia200x** [comment]
implementation of KConfig files looks also good. It should be easy to proceed.

**Comment by miri64** [comment]
> @miri64 do you have something to add?

Nope all good from my side. Murdock seems to have failed on an unrelated error though.

**Pull request merged**